Dependencies/xr/Source/OpenXR/XR.cpp

```diff
@@ -1573,101 +1610,132 @@ namespace xr
-                    {
-                        // map the openxr values to populate other states of a button and axes that webxr expects
-                        gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Pressed = (gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Value == 1);
-                        gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Touched = (gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Value > 0);
-                        gamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Value = (gamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Pressed);
-                        gamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Touched = (gamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Pressed);
-                        gamepadObject.Axes[controllerInfo.TRACKPAD_Y_AXIS] = -(gamepadObject.Axes[controllerInfo.TRACKPAD_Y_AXIS]);
-                        gamepadObject.Buttons[controllerInfo.TRACKPAD_BUTTON].Value = (gamepadObject.Buttons[controllerInfo.TRACKPAD_BUTTON].Pressed);
-                        gamepadObject.Axes[controllerInfo.THUMBSTICK_Y_AXIS] = -(gamepadObject.Axes[controllerInfo.THUMBSTICK_Y_AXIS]);
-                        gamepadObject.Buttons[controllerInfo.THUMBSTICK_BUTTON].Value = (gamepadObject.Buttons[controllerInfo.THUMBSTICK_BUTTON].Pressed);
-                        gamepadObject.Buttons[controllerInfo.THUMBSTICK_BUTTON].Touched = (gamepadObject.Axes[controllerInfo.THUMBSTICK_X_AXIS] != 0 || gamepadObject.Axes[controllerInfo.THUMBSTICK_Y_AXIS] != 0);
-
-                        // Only signal that gamepad data is available if the actions were available
-                        InputSources[idx].GamepadTrackedThisFrame = true;
-                    }
-                }
-
-                // Get joint data
-                if (sessionImpl.HandData.HandsInitialized)
-                {
-                    if (sessionImpl.HmdImpl.Context.Extensions()->HandInteractionSupported)
+                    auto& inputSource = InputSources[idx];
+                    XrPath interactionProfilePath;
+                    XrCheck(xrStringToPath(m_impl->sessionImpl.HmdImpl.Context.Instance(), inputSource.interactionProfileName.data(), &interactionProfilePath));
+                    switch (interactionProfilePath)
                     {
-                        const auto& controllerInfo = sessionImpl.ControllerInfo;
-                        auto& gamepadObject = InputSources[idx].GamepadObject;
-
-                        // Get interaction data
-                        if ((m_impl->TryUpdateControllerBooleanAction(actionResources.HandGetSelectAction, session, gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Pressed)) &&
-                            (m_impl->TryUpdateControllerBooleanAction(actionResources.HandGetSqueezeAction, session, gamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Touched)))
+                        case 31:// MICROSOFT_XR_INTERACTION_PROFILE
```

Comment:
>31 [](start = 29, length = 2)

nit: do we have constants declaring these case values? if not, these comments are good

Reply:
I searched and couldn't find any. I'd really rather use constants, though.